--- spheno.f90	2009-10-15 13:01:57.000000000 +0200
+++ spheno.f90	2009-10-15 13:04:28.000000000 +0200
@@ -12,7 +12,7 @@
 !     the email above, if possible with a copy of the screen output 
 ! written by Werner Porod
 !-----------------------------------------------------------------
-Program SPheno
+Module SPhenoDouble
 
 !---------------------------------
 ! loading necessary modules
@@ -63,6 +63,10 @@
  Integer :: kont, i1, i_min(3)
  Real(dp) :: m_Gut, ratioWoM
 
+
+Contains
+
+ Subroutine RunTill_Model_bilinear_Rparity
  !--------------------------------------------------------------------------
  ! set all parameters to zero
  !--------------------------------------------------------------------------
@@ -179,6 +183,10 @@
 
  End If
 
+ End Subroutine RunTill_Model_bilinear_Rparity
+
+
+ Subroutine RunTill_End
  If ((kont.Eq.0).And.(HighScaleModel.Ne."NMSSM")) Then
 
   Call CalculateLowEnergyConstraints(gauge, Y_l, Y_d, Y_u      &
@@ -232,8 +240,7 @@
 
  Call closing() ! closes the files
 
-
-Contains
+ End Subroutine RunTill_End
 
 
  Subroutine CalculateLowEnergyConstraints(gauge, Y_l, Y_d, Y_u, mSpm2, RSpm   &
@@ -1382,5 +1389,5 @@
 
  End Subroutine ReadingData
 
-End Program SPheno
+End Module SPhenoDouble
 
